dev/src/reference.ts

```
@@ -1399,17 +1568,29 @@ export class Query<T = firestore.DocumentData> implements firestore.Query<T> {
       }
     }
 
-    const fieldFilter = new FieldFilter(
+    return new FieldFilterInternal(
       this._serializer,
       path,
-      comparisonOperators[opStr],
+      comparisonOperators[operator],
       value
     );
+  }
 
-    const options = this._queryOptions.with({
-      fieldFilters: this._queryOptions.fieldFilters.concat(fieldFilter),
-    });
-    return new Query(this._firestore, options);
+  _parseCompositeFilter(compositeFilterData: CompositeFilter): FilterInternal {
+    const parsedFilters = compositeFilterData
+      ._getFilters()
+      .map(filter => this._parseFilter(filter))
+      .filter(parsedFilter => parsedFilter.getFilters().length > 0);
+
+    // For composite filters containing 1 filter, return the only filter.
+    // For example: AND(FieldFilter1) == FieldFilter1
+    if (parsedFilters.length === 1) {
+      return parsedFilters[0];
+    }
+    return new CompositeFilterInternal(
+      parsedFilters,
+      compositeFilterData._getOperator() === 'AND' ? 'AND' : 'OR'
```

Should we throw an error here instead if `OR` or `AND` is not provided?